app/src/main/res/values/strings.xml

```diff
@@ -239,10 +239,18 @@
   <string name="upload_image_problem_dark">This picture is too dark, are you sure you want to upload it? Wikimedia Commons is only for pictures with encyclopedic value.</string>
   <string name="upload_image_problem_blurry">This picture is blurry, are you sure you want to upload it? Wikimedia Commons is only for pictures with encyclopedic value.</string>
+  <string name="upload_image_problem_wrong_location">This picture is taken at another location, are you sure you want to upload for this place?</string>
+  <string name="upload_image_problem_wrong_location_dark">This picture is too dark and taken at another location, are you sure you want to upload for this place? Wikimedia Commons is only for pictures with encyclopedic value.</string>
+  <string name="upload_image_problem_wrong_location_duplicate">This picture is already exists on Commons and taken at another location, are you sure you want to upload for this place? Wikimedia Commons is only for pictures with encyclopedic value.</string>
+  <string name="upload_image_problem_wrong_location_blurry">This picture is blurry and taken at another location, are you sure you want to upload for this place? Wikimedia Commons is only for pictures with encyclopedic value.</string>
   <string name="upload_image_problem_dark_duplicate">This picture is too dark and already exists on Commons, are you sure you want to upload it? Please only upload pictures with encyclopedic value.</string>
+  <string name="upload_image_problem_dark_duplicate_wrong_location">This picture is too dark, duplicate and taken at another location, are you sure you want to upload it, for this place?</string>
   <string name="upload_image_problem_blurry_duplicate">This picture is blurry and already exists on Commons, are you sure you want to upload it? Please only upload pictures with encyclopedic value.</string>
+  <string name="upload_image_problem_blurry_duplicate_wrong_location">This picture is blurry, duplicate and taken at another location, are you sure you want to upload it, for this place?</string>
   <string name="upload_image_problem_dark_blurry">This picture is too dark and is blurry, are you sure you want to upload it? Please only upload pictures with encyclopedic value.</string>
+  <string name="upload_image_problem_dark_blurry_wrong_location">This picture is too dark, blurry and is taken at another location, are you sure you want to upload it, for this place?</string>
   <string name="upload_image_problem_dark_blurry_duplicate">This picture is too dark, is blurry and already exists on Commons, are you sure you want to upload it? Please only upload pictures with encyclopedic value.</string>
+  <string name="upload_image_problem_dark_blurry_duplicate_wrong_location">This picture is too dark, blurry, duplicate and is taken at another location, are you sure you want to upload it, for this place?</string>
```

Comment:
Hmmm. I am not sure how I feel about combining all of these warnings. Making any modifications to the checks is going to be tricky with all of these permutations. And every check we add will create (x+1)! permutations that need to be handled! What about just letting the dialogs trigger one by one and just having 1 dialog for each checked case? @maskaravivek @nicolas-raoul

Reply:
I do think we shouldn't increase them more. On the other hand I think displaying them with seperate dialogs could be disturbing. 

When it comes to current implementation, there was a really neat implementation, I couldn't stop myself from joining:) And there is no uncovered case in current implementation. For further cases, I do think we should improve our method.

Reply:
Well the selfie check will be coming up if our PG is approved. :) But, it is not just about adding. For instance, if we change or remove anything in the phrase, it would require the modification of 5 strings instead of just one. A trivial effort for us to find and replace, perhaps, but then translators would have to manually change all 5 strings...

Reply:
How about a single popup that just lists the warnings? For instance:
```
Blabla selfie blabla sad.
Blabla dark blabla sad.
Blabla location blabla sad.
```

Reply:
I agree to @nicolas-raoul 's idea, and very easy to adapt current implementation:)